registration/app/controllers/Main.scala

```diff
@@ -3,49 +3,29 @@ package controllers
 import javax.inject.Inject
 
 import gu.msnotifications.HubFailure.{HubInvalidConnectionString, HubServiceError, HubParseFailed}
-import gu.msnotifications._
 import models.{Registration, ApiResponse, WindowsMobile}
 import play.api.Logger
 import play.api.libs.json.Json
 import play.api.mvc.{Action, BodyParsers, Controller, Result}
 import services._
-import scala.async.Async
 import scala.concurrent.{Future, ExecutionContext}
-import scalaz.{-\/, \/-}
+import scalaz.{\/, -\/, \/-}
 import BodyParsers.parse.{json => BodyJson}
+import notifications.providers.{Error => ProviderError}
```

Comment:
If it's too generic why not to rename it in place i.e. in its own definition ? ;)